src/routes/PaymentCheckPage.tsx

```diff
@@ -134,19 +134,23 @@ export default function PaymentCheckPage() {
     window.location.replace(authorizationUrl);
   };
 
-  const stDate = "01/".concat(cardData?.expDate || "");
-  const pattern = /(\d{2})\/(\d{2})\/(\d{2})/;
-  const dateParsed = new Date(stDate.replace(pattern, "$2-$1-$3"));
+  const dateString = "01/".concat(cardData?.expDate || "");
+  const datePattern = /(\d{2})\/(\d{2})\/(\d{2})/;
+  const [day, month, year] = (dateString.match(datePattern) || []).slice(1);
+  const dateParsed = new Date(Number(year)+2000, Number(month) - 1, Number(day));
```

Comment:
You could make it a little more safe (I haven't checked the surrounding context to see how the date is used btw)

```suggestion
  const dateString = pipe(
      O.fromNullable(cardData?.expDate),
      O.map(d => "01/".concat(d))
  );
  const datePattern = /(\d{2})\/(\d{2})\/(\d{2})/;

  const date = pipe(
      dateString,
      O.flatMap(date => {
          return O.fromNullable(date.match(datePattern))
      }),
      O.map(matches => matches.slice(1)),
      O.map(matches => [matches[0], matches[1], matches[2]]),
      O.map(([day, month, year]) => new Date(Number(year) + 2000, Number(month) - 1, Number(day))),
      O.getOrElse(null) // Only if there is no other choice :)
  );
```